internal/sbi/producer/generate_auth_data.go

```diff
@@ -394,7 +395,18 @@ func GenerateAuthDataProcedure(authInfoRequest models.AuthenticationInfoRequest,
 			sqnStr = fmt.Sprintf("%x", bigSQN)
 			sqnStr = strictHex(sqnStr, 12)
 		} else {
-			logger.UeauLog.Errorln("Re-Sync MAC failed ", supi)
+			logger.UeauLog.Errorln("Re-Sync MAC failed ", supiOrSuci)
+			// Check if suci
+			if suciPart := strings.Split(supiOrSuci, "-"); suciPart[0] == "suci" {
+				// Get SuciProfile index and write public key
+				keyIndex, err := strconv.Atoi(suciPart[suci.HNPublicKeyIDPlace])
```

Comment:
need to rename `err` (ex: `err1`) to prevent golangci-linter fail: `shadow: declaration of "err" shadows declaration at line 158 (govet)`